src/core/stream/adaptation/adaptation_stream.ts

```diff
@@ -481,15 +482,22 @@ export default function AdaptationStream(
   }
 
   /**
-   * @param {Object} representation
+   * Returns how much media data should be pre-buffered for this
+   * `Representation`, according to the `wantedBufferAhead` setting and previous
+   * issues encountered with that `Representation`.
+   * @param {Object} representation - The `Representation` you want to buffer.
+   * @param {number} wba - The value of `wantedBufferAhead` set by the user.
    * @returns {number}
    */
-  function getBufferGoalRatio(representation: IRepresentation): number {
+  function getBufferGoal(representation: IRepresentation, wba: number): number {
     const oldBufferGoalRatio = bufferGoalRatioMap.get(representation.id);
     const bufferGoalRatio = oldBufferGoalRatio !== undefined ? oldBufferGoalRatio : 1;
     if (oldBufferGoalRatio === undefined) {
       bufferGoalRatioMap.set(representation.id, bufferGoalRatio);
     }
-    return bufferGoalRatio;
+    if (bufferGoalRatio < 1 && wba === Infinity) {
+      return 5 * 60 * 1000 * bufferGoalRatio;
```

Comment:
maybe we can add a comment to say that 5 minutes is the default for calculation in case wanted buffer head is infinite

Reply:
Done